mappings/ucl/cause-of-death-mapping.json

```diff
@@ -1,56 +1,65 @@
+      },
+      "expression": {
+        "name": "result",
+        "language": "application/fhir-template+json",
+        "value": {
+          "id": "{{mpp:getHashedId('Observation', 'Death' & person_id.toString())}}",
+          "meta": {
+            "profile": [
+              "https://datatools4heart.eu/fhir/StructureDefinition/HFR-CauseOfDeath"
+            ]
+          },
+          "resourceType": "Observation",
+          "status": "final",
+          "category": [
+            {
+              "coding": [
+                {
+                  "system": "http://terminology.hl7.org/CodeSystem/observation-category",
+                  "code": "exam",
+                  "display": "Exam"
+                }
+              ]
+            }
+          ],
+          "code": {
             "coding": [
               {
-                "system": "http://terminology.hl7.org/CodeSystem/observation-category",
-                "code": "exam",
-                "display": "Exam"
+                "system": "http://loinc.org",
+                "code": "79378-6",
+                "display": "Cause of death",
+                "@sliceName": "loincCode"
               }
             ]
-          }
-        ],
-        "code" : {
-          "coding" : [ {
-            "system" : "http://loinc.org",
-            "code" : "79378-6",
-            "display" : "Cause of death",
-            "@sliceName" : "loincCode"
-          } ]
-        },
-        "effectiveDateTime" : "{{iif(death_datetime.empty(), death_date, death_datetime)}}",
-        "valueCodeableConcept" : {
-          "coding" : "{{* iif(cause_source_value.empty() or cause_source_value = 'NA', trms:translateToCoding(cause_concept_id.toString(), 'https://www.ohdsi.org/omop', 'https://datatools4heart.eu/fhir/mappings/ConceptMap/snomed-to-icd10').first(), utl:createFhirCoding('http://hl7.org/fhir/sid/icd-10', cause_source_value, {}))}}"
-        },
-        "subject" : "{{mpp:createFhirReferenceWithHashedId('Patient', person_id.toString())}}",
-        "id" : "{{mpp:getHashedId('Observation', 'Death' & person_id.toString())}}",
-        "meta" : {
-          "profile" : [ "https://datatools4heart.eu/fhir/StructureDefinition/HFR-CauseOfDeath" ]
-        },
-        "resourceType" : "Observation"
+          },
+          "effectiveDateTime": "{{iif(death_datetime.empty(), death_date, death_datetime.utl:toFhirDateTime('yyyy-MM-dd HH:mm:ss.SSS'))}}",
+          "valueCodeableConcept": {
+            "coding": "{{* iif(cause_concept_id.exists(),trms:translateToCoding(cause_concept_id.toString(), 'https://www.ohdsi.org/omop', 'https://datatools4heart.eu/fhir/mappings/ConceptMap/omop-to-icd10').first(),iif(cause_source_concept_id.exists(), trms:translateToCoding(cause_source_concept_id.toString(), 'https://www.ohdsi.org/omop', 'https://datatools4heart.eu/fhir/mappings/ConceptMap/omop-to-icd10').first() , {}))}}"
```

Comment:
I believe we should put all ICD-10 codes retrieved from the terminology mapping into this Observation resource. And, we need to check with some kind of "contains" function while developing the feature definitions.

Reply:
I would suggest to do the same for all "first()" functions that are called on terminology translation results.

Reply:
I've updated all terminology service usages in the UCL mappings, and everything looks good except for cause-of-death-mapping.json.

The issue is that we can't include all relevant ICD-10 codes due to a max = 1 restriction in the profile. If you're okay with lifting this restriction, I can update the profile accordingly and proceed with the feature extraction.

Reply:
Yes, I am OK with lifting this restriction.

Reply:
This has been completed, but please note that the update was applied only to the UCL mappings. I've made a note to update the remaining mappings separately in their respective branches at a later stage. Resolving the conversation.